document/core/syntax/instructions.rst

```diff
@@ -344,6 +349,35 @@ Some SIMD instructions have a signedness annotation |sx| which distinguishes whe
 For the other SIMD instructions, the use of two's complement for the signed interpretation means that they behave the same regardless of signedness.
 
 
+Conventions
+...........
+
+.. todo::
+   the macros here refer to the numeric instruction, we might want macros that refer to simd instructions of the same name
+
+Occasionally, it is convenient to group operators together according to the following grammar shorthands:
+
+.. math::
+   \begin{array}{llll}
+   \production{unary operator} & \vunop &::=&
+     \viunop ~|~
+     \vfunop \\&&|&
+     \NEG ~|~
+     \WIDEN \\
+   \production{binary operator} & \vbinop &::=&
+     \vibinop ~|~ \vfbinop \\&&|&
+     \virelop ~|~ \vfrelop \\&&|&
+     \viminmaxop ~|~ \visatbinop \\&&|&
+     \SWIZZLE ~|~
+     \NARROW ~|~
+     \MUL ~|~
+     \AVGRU ~|~
+     \TRUNC ~|~
+     \CONVERT \\
+   \production{ternary operator} & \vsternop &::=& \SHUFFLE \\
```

Comment:
Why is this vsternop while the others are vunop, vbinop (without the s)? Are vsunop, vsbinop defined somewhere else? (They appear in validation rules.)

Reply:
My horrible naming convention is:
- vs* is all the shape-agnostic instructions, like v128.and
- vi* is all the int-shape instructions, like i32x4.add
- vf* is all the float-shape instructions, like f32x4.sqrt
- v* is union of vi* and vf*

vsunop and vsbinop are defined in the production for instructions above, they aren't grouped as part of this "Conventions" section.

I decided to not group them together becase the validation instructions have 2 different matches, `t.op` which is for `vs*` family of instructions, and `vxx.op`, for the vi*, vf*, v* family.(since `t` matches `v128` and `vxx` matches all the shapes.)

Reply:
Hm, but `vsternop` is already defined earlier as
```
vsternop ::= bitselect
```
So there's some clash here. Shouldn't the one defined here rather be named `vternop`?

Also, wouldn't it be more consistent if `vtestop` was `vitestop`?

Reply:
Yea you're right, the `vsternop ::= shuffle` here is wrong, shuffle is not a ternop. I'll remove this.
Also will make the change from `vtestop` to `vitestop`.